service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -278,4 +295,42 @@ void getDatasetPreviewTable() {
         datasetPreviewTable.getColumns().get(0),
         is(new bio.terra.catalog.model.ColumnModel().name("column a").arrayOf(false)));
   }
+
+  @Test
+  void testExportDatarepoDataset() {
+    reset(datasetDao);
+    when(datasetDao.retrieve(datasetId)).thenReturn(tdrDataset);
+    when(datarepoService.getRole(user, sourceId)).thenReturn(DatasetAccessLevel.READER);
+    String workspaceId = String.valueOf(UUID.randomUUID());
+    doThrow(new BadRequestException("error"))
+        .when(datarepoService)
+        .exportDatarepoDataset(user, sourceId, workspaceId);
+    assertThrows(
+        BadRequestException.class,
+        () -> datasetService.exportDataset(user, datasetId, workspaceId));
+  }
+
+  @Test
+  void testExportExternalDataset() {
```

Comment:
This test doesn't appear to be specific to external datasets. It's verifying that `exportDataset()` is doing a permission check, which fails because the current user isn't an admin user. If you mock that in sam, then you can write a test that verifies that trying to export an external dataset is a no-op.
```
    when(samService.hasGlobalAction(user, SamAction.READ_ANY_METADATA)).thenReturn(true);
```

But I wouldn't recommend doing that as we haven't worked on external datasets yet so there's not much value to us to verify our APIs behavior when using external datasets.

So you could keep this test but make it use either the TDR or workspace dataset.

Reply:
I removed this test.